### PR TITLE
fixes duplicate list items on welcome #11366 #11578

### LINKED ISSUE #11366: Recent File List Entries duplicating list items
Steps to Reproduce:

1. Open Azure Data Studio
2. If welcome page not open, open welcome page
3. Under resents notice the duplication of files

### LINKED ISSUE #11578: Welcome page history shows a bunch of items the shrinks on startup
- Azure Data Studio Version: `main`

Steps to Reproduce:

1. Open ADS with Welcome page open and lots of items in history

Expected: History loads with a particular view and doesn't change after loading

Actual: History list shows all the items and then a second or two later reduces to a list of a few items and a "Show More" link.  The UI change after welcome page is done load is pretty noticeable.

### PR DESCRIPTION
**Pull request by v-bbrady**
This PR fixes the bug that when the welcome page is first opened the recent files list duplicates file listings for a moment, and then promptly removes the unwanted list items. However now the welcome page loads with the appropriate list items.

This PR fixes #11366
This PR fixes #11578

### CODE REVIEW

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 1]
```diff
@@ -680,25 +677,30 @@ class WelcomePage extends Disposable {
...
-			let fullPath: URI;
-			let windowOpenable: IWindowOpenable;
-			if (isRecentFolder(recent)) {
-				windowOpenable = { folderUri: recent.folderUri };
-				relativePath = recent.label || this.labelService.getWorkspaceLabel(recent.folderUri, { verbose: true });
-				fullPath = recent.folderUri;
-			} else {
-				relativePath = recent.label || this.labelService.getWorkspaceLabel(recent.workspace, { verbose: true });
-				windowOpenable = { workspaceUri: recent.workspace.configPath };
+		if (!this.called) {
```
Comment (alanrenmsft) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:682]: I think we don't need this 'called' flag, if we clear the items before rendering the recent list we should be good.

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 2]
```diff
@@ -680,25 +677,30 @@ class WelcomePage extends Disposable {
...
+				}
+				const elements = await this.createListEntries(container, fileService, fullPath, windowOpenable, relativePath);
+				result.push(...elements);
 			}
-			const elements = await this.createListEntries(container, fileService, fullPath, windowOpenable, relativePath);
-			result.push(...elements);
+			return result;
 		}
-		return result;
+		return null;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:703]: `undefined` instead of `null`

**Comment by Charles-Gagnon** [comment]
@v-bbrady Code complete is today - are you planning on finishing up this PR before then?

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 3]
```diff
@@ -316,9 +316,15 @@ class WelcomePage extends Disposable {
 			}
 			const workspacesToShow = workspaces.slice(0, 5);
 			const updateEntries = async () => {
-				await this.mapListEntries(workspacesToShow, fileService, container);
+				while (ul.firstChild) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:317]: I'd suggest just using clearNode from https://github.com/microsoft/azuredatastudio/blob/main/src/vs/base/browser/dom.ts#L20

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 4]
```diff
@@ -316,9 +316,15 @@ class WelcomePage extends Disposable {
...
+					ul.removeChild(ul.firstChild);
+					await this.mapListEntries(workspacesToShow, fileService, container);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:319]: Why are you doing these calls in every iteration of the loop? Don't you want to just do it once after the node has been cleared?

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 5]
```diff
@@ -308,19 +308,16 @@ class WelcomePage extends Disposable {
...
 			if (!ul) {
 				return;
 			}
 			const workspacesToShow = workspaces.slice(0, 5);
-			const updateEntries = async () => {
-				while (ul.firstChild) {
-					ul.removeChild(ul.firstChild);
-				}
-				await this.mapListEntries(workspacesToShow, fileService, container);
+			const updateEntries = () => {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:316]: You're no longer calling this onDidChangeFormatters - is that intentional? If you don't need to call this again then just get rid of the function and call these directly.

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 6]
```diff
@@ -308,19 +308,16 @@ class WelcomePage extends Disposable {
...
+				clearNode(ul);
+				this.mapListEntries(workspacesToShow, fileService, container, ul);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:318]: Purposely not awaiting this call anymore?

Comment (v-bbrady) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts]: That was unintentional, fixed.

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts (hunk 5 above)
Comment (v-bbrady) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts]: fixed

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 7]
```diff
@@ -308,19 +308,13 @@ class WelcomePage extends Disposable {
...
 			const workspacesToShow = workspaces.slice(0, 5);
-			const updateEntries = async () => {
-				while (ul.firstChild) {
-					ul.removeChild(ul.firstChild);
-				}
-				await this.mapListEntries(workspacesToShow, fileService, container);
-			};
-			await updateEntries();
-			this._register(this.labelService.onDidChangeFormatters(updateEntries));
+			clearNode(ul);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:316]: So is clearing this actually needed anymore? I would expect onReady to only ever be called once right?

**Comment by alanrenmsft** [comment]
@v-bbrady as well discussed offline last Friday, have you figured out why is it being called more than once?

**Pull request merged**